AK/Time.cpp

```diff
@@ -38,6 +38,15 @@ unsigned day_of_week(int year, unsigned month, int day)
     return (year + year / 4 - year / 100 + year / 400 + seek_table[month - 1] + day) % 7;
 }
 
+Time Time::from_ticks(clock_t ticks, time_t ticks_per_second)
+{
+    auto secs = ticks % ticks_per_second;
+
+    i32 nsecs = 1'000'000'000 * (ticks - (ticks_per_second * secs)) / ticks_per_second;
```

Comment:
This looks extremely likely to overflow when using `i32`. Any reason to not use `i64` here?

Reply:
AFAICT the computation needs to be done on 64 bits, but the result will always fit on 32 bits.
I'm not entirely sure, but this should be good with integer promotions.
```ticks - (ticks_per_second * secs)``` will always be one power of then smaller than ```ticks_per_second``` so ```ticks - (ticks_per_second * secs) / ticks_per_second``` will be lower than 1. By the end i32 is smaller than ```1e9``` i.e. doesn't overflow.